workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/IMemberAdapter.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Reflection;
+
+namespace Improbable.Gdk.Core
+{
+    /// <returns>
+    ///     A handle on a field/property of a type obtained through reflection, for getting/setting it on instances conveniently.
+    /// </returns>
+    public interface IMemberAdapter
```

Comment:
I'm not sure what the alternative is that you're suggesting, is it just to have the types more specific, setting IComponentReaders on MonoBehaviours? If that's it then yes, that should work fine too

Reply:
I believe this approach was carried over from the Unity SDK?
I think having these is probably over-engineering.

Reply:
It's useful for handling fields and properties together. Perhaps not an important thing to have, but the SDK supported it and since we have this implementation it's not a hard thing to have that supported right away.